src/layouts/BaseLayout/BaseLayout.jsx

```diff
@@ -36,11 +35,7 @@ const OnboardingOrChildren = ({ children }) => {
   const { isImpersonating } = useImpersonate()
 
   if (showAgreeToTerms && !isFullExperience) {
-    return (
-      <Suspense fallback={null}>
-        <TermsOfService />
-      </Suspense>
-    )
+    return <Redirect to="/terms" />
```

Comment:
I think having a dedicated route and using redirects means the original path the user was trying to access will be lost. Example:

User comes from a PR link, hasn't signed TOS, when they sign it they're sent to the `/:provider/:org` not the PR they clicked on.

https://github.com/codecov/gazebo/pull/2321/files#diff-a8fba335305fed7cd0a638ac77db50888a5c94a8c0115d5cd78a00664904dec2R112


Is this a known trade off we're making?

Reply:
I think it’s fine. we require no synced services to render this page, TOS lived under a component that used queries that serve on providers so it made sense to take it out, otherwise we would have to alter HomePageRedirect
and other components to make this happen

Reply:
Ps. Might be worth it to do the same thing for default org? sync page has it’s own dedicated route as well, I think it’s weird to jump from `/terms` -> `/sync` -> `/` But then again, default org works fine with services.

Reply:
I've to raised this to AJ. I (unfortunately) think loosing the navigation path is not good UX and that another approach might be needed here.. But I'll let him weigh the trade offs

Reply:
sure. do you mind including me in the conversation please?

Reply:
Yup for sure! I directed him to this issue actually how ever if more slacking happens we'll move to the team channel

Reply:
thank you